packages/core/helper-plugin/src/hooks/useSelectionState.ts

```diff
@@ -1,9 +1,13 @@
 import { useState } from 'react';
 
-export const useSelectionState = (keys, initialValue) => {
+type JSON = string | number | boolean | null | { [key: string]: JSON } | Array<JSON>;
+
+export type InitialValue = Record<string, JSON>;
+
+export const useSelectionState = (keys: string[], initialValue: InitialValue[]) => {
```

Comment:
ohh yes right 👍

Reply:
because the dynamic value in all this is the shape of the object we're passing in the array that we're additionally indexing, because we're using `keyof` we know `TValues` should extend the object type :)